### PR TITLE
Separate out linux & windows implementation of CollectSystemInfo()

### PR DESCRIPTION
**Pull request by derekwbrown**
(do not merge yet.  Needs associated gohai change to be merged first)

On windows, don't use WMI, use gohai implementation that collects
same information (and can gracefully fail when some info isn't
available, e.g. in a container) .

Allows process agent to start in a Nano container.

Requires related gohai change.

### Motivation

allow process agent to run on Nano

### CODE REVIEW

File: pkg/process/checks/system_info_windows.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+func CollectSystemInfo(cfg *config.AgentConfig) (*model.SystemInfo, error) {
+	hi, err := platform.GetArchInfo()
+	if err != nil {
+		return nil, err
+	}
+	cpuInfo, err := cpu.GetCpuInfo()
+	if err != nil {
+		return nil, err
+	}
+	fmt.Printf("cpu info %v\n", cpuInfo)
```
Comment (albertvaka) [inline pkg/process/checks/system_info_windows.go:27]:
```suggestion
```

File: pkg/process/checks/system_info_windows.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	l2Cache, _ := strconv.ParseInt(cpuInfo["cache_size_l2"], 10, 64)
+	cpus := make([]*model.CPUInfo, 0)
+	for i := int64(0); i < physCount; i++ {
+		cpus = append(cpus, &model.CPUInfo{
+			Number:     int32(i),
+			Vendor:     cpuInfo["vendor_id"],
+			Family:     cpuInfo["family"],
+			Model:      cpuInfo["model"],
+			PhysicalId: "",
+			CoreId:     "",
```
Comment (albertvaka) [inline pkg/process/checks/system_info_windows.go:44]: Is it okay to lose these fields?

File: pkg/process/checks/system_info_windows.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	mi, err := winutil.VirtualMemory()
+	if err != nil {
+		return nil, err
+	}
+	physCount, _ := strconv.ParseInt(cpuInfo["cpu_pkgs"], 10, 64)
+	logicalCount, _ := strconv.ParseInt(cpuInfo["cpu_logical_processors"], 10, 64)
+	clockSpeed, _ := strconv.ParseInt(cpuInfo["mhz"], 10, 64)
+	l2Cache, _ := strconv.ParseInt(cpuInfo["cache_size_l2"], 10, 64)
+	cpus := make([]*model.CPUInfo, 0)
+	for i := int64(0); i < physCount; i++ {
```
Comment (albertvaka) [inline pkg/process/checks/system_info_windows.go:37]: This assumes all CPUs installed are equal, which might not be the case. Can this be a problem?

Comment (derekwbrown) [reply pkg/process/checks/system_info_windows.go:37]: If you have multiple CPUs they have to be equal.

File: pkg/process/checks/system_info_windows.go (hunk 2 above)
Comment (derekwbrown) [reply pkg/process/checks/system_info_windows.go:43]: those weren't being filled in by the underlying `gopsutil` anyway

**Review by albertvaka — DISMISSED** [review]

**Review by albertvaka — APPROVED** [review]

**Review by shang-wang — APPROVED** [review]

**Pull request merged**